Unit tests fail to compile on macOS: ambiguous TextStream::operator<< in scorecomp.cpp

### Issue type
Other type of issue (build / compile error)

### Description with steps to reproduce
Building the unit tests on macOS fails to compile `src/engraving/tests/utils/scorecomp.cpp`. It compiles on Linux.

Steps:
1. On macOS (Apple Clang), configure with `MUSE_ENABLE_UNIT_TESTS=ON`.
2. Build the `engraving_tests` target.

Actual: compilation fails with
```
src/engraving/tests/utils/scorecomp.cpp:219:22: error: use of overloaded operator '<<' is ambiguous (with operand types 'TextStream' and 'const size_t' (aka 'const unsigned long'))
muse/framework/global/serialization/textstream.h:52:17: note: candidate function: TextStream& operator<<(int32_t)
muse/framework/global/serialization/textstream.h:53:17: note: candidate function: TextStream& operator<<(uint32_t)
muse/framework/global/serialization/textstream.h:55:17: note: candidate function: TextStream& operator<<(int64_t)
muse/framework/global/serialization/textstream.h:56:17: note: candidate function: TextStream& operator<<(uint64_t)
```
The same error repeats on lines 221, 223 and 225.

Expected: the unit tests compile on macOS, as they do on Linux.


### In which versions of MuseScore Studio is this issue present?
Current `main` development build (5.0.0), since PR #33822. Not present in released 4.x.

### Regression
Yes, this works in the latest released version and is broken in the nightly builds.

### Operating system
macOS 26.5.1 (Apple Clang, arm64)

### Additional context
The unit tests are compiled only by `check_unit_tests.yml` on `ubuntu-24.04`; the macOS job builds the app bundle with unit tests disabled, so this file is never compiled on macOS in CI. Introduced by PR #33822 (`ebcbb5eba6`).

### Checklist
- [x] This report follows the guidelines for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the issue tracker for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

--- a/src/engraving/tests/utils/scorecomp.cpp
+++ b/src/engraving/tests/utils/scorecomp.cpp
@@ -216,13 +216,13 @@ bool ScoreComp::compareFiles(const String& fullPath1, const String& fullPath2)
         const size_t headerStart1 = start1Set ? start1 + 1 : 0;
         const size_t headerStart2 = start2Set ? start2 + 1 : 0;
 
-        ts << "@@ -" << headerStart1;
+        ts << "@@ -" << static_cast<uint64_t>(headerStart1);
         if (count1 != 1) {
-            ts << "," << count1;
+            ts << "," << static_cast<uint64_t>(count1);
         }
-        ts << " +" << headerStart2;
+        ts << " +" << static_cast<uint64_t>(headerStart2);
         if (count2 != 1) {
-            ts << "," << count2;
+            ts << "," << static_cast<uint64_t>(count2);
         }
         ts << " @@\n";
 

